### PR TITLE
Bug 1672691 - part 2:  [beetmoverscript] Add integration tests to beetmover (more cases)

### PR DESCRIPTION
**Pull request by alopezz**
This adds more test cases for integration testing on beetmover.

In a first draft, I'm adding a test for geckoview-nightly, moving task data to a separate folder and reorganized helper functions to make it easier to reuse common setup and assertion logic.

Would it be possible to get a list of possible cases worth adding so that I can build some sort of checklist to have a better of idea of when I'm done?

### CODE REVIEW

**Comment by JohanLorenzo** [comment]
Cool! I tried to think of a test suite that covers most cases while not repeating too much. In addition to the ones you already found, here's what I got:

* Firefox desktop nightly: https://firefox-ci-tc.services.mozilla.com/tasks/JD3Pxn_HRj6YxbqEfpKchw
* Firefox langpacks beta: https://firefox-ci-tc.services.mozilla.com/tasks/JErgjPZ8QqiwMxG91lnm8g
* Firefox generated screenshots release: https://firefox-ci-tc.services.mozilla.com/tasks/bAGO1prJQESNjKjmFbtaIQ
* Application-services release: https://firefox-ci-tc.services.mozilla.com/tasks/MshCrl1wQuu66B8SF2W60A
* Glean release: https://firefox-ci-tc.services.mozilla.com/tasks/DTo21kh1TAygClAqpHXJTg

That may be a lot to handle in a single PR. Feel free to split it in multiple ones, if you think it's worth it.

**Comment by alopezz** [comment]
Splitting into multiple PRs makes sense to me as well and should make it easier to review. I'll group all the `push-to-maven` tasks into this first PR.

Do we have examples that would trigger `push-to-partner` and `push-to-releases` actions? (as taken from the `scopes` field in the task definition)

**Comment by JohanLorenzo** [comment]
Oh good point! Thanks for catching this 😃

* `push-to-partner`: https://firefox-ci-tc.services.mozilla.com/tasks/MYKL7DcHT4qO8xdGH8DdTQ 
* `push-to-releases`: https://firefox-ci-tc.services.mozilla.com/tasks/G40GsqdZQR6k7ZQWnqiheQ

**Comment by alopezz** [comment]
All four proposed `push-to-maven` task cases are now covered, so this PR can be merged if reviewed favourably.

File: beetmoverscript/tests/test_integration.py [hunk 1]
```diff
@@ -232,11 +131,24 @@ def scriptworker_config(tmp_path, config, task):
...
+        ("android-components", "android_components_nightly", 12),
+        ("geckoview", "geckoview_nightly", 16),
+        ("app-services", "application_services_release", 12),
+        ("glean", "glean_release", 20),
+    ),
+)
+def test_main_push_to_maven(config_name, task_name, expected_number_of_artifacts, tmp_path, boto3_client_mock, aiohttp_session_mock):
+    config = get_config(config_name)
+    task = get_test_task(task_name)
+    scriptworker_config = prepare_scriptworker_config(tmp_path, config, task)
```
Comment (JohanLorenzo) [inline beetmoverscript/tests/test_integration.py:146]: Note to self: `scriptworker_config` is not a fixture anymore because of `get_config(config_name)`

**Review by JohanLorenzo — APPROVED** [review]
Awesome! I'm super happy to get this merged. Great job! 💯

File: beetmoverscript/tests/test_integration.py (hunk 1 above)
Comment (alopezz) [reply beetmoverscript/tests/test_integration.py:146]: Yeah! The need to parametrize the tests (and needing to implement the functionality of `scriptworker_config` for all cases) led to changing the approach like that.

**Pull request merged**